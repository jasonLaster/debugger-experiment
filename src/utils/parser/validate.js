// @flow

import { parseExpression } from "./utils/ast";

export function hasSyntaxError(input: string) {
  try {
    parseExpression(input, { sourceType: "script" });
    return false;
  } catch (e) {
    return `${e.name} : ${input} is an invalid.`;
  }
}
