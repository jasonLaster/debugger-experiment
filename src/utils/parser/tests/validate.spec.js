import { validateSyntax } from "../validate";

describe("validate syntax", () => {
  it("should return the  valid expression", () => {
    expect(validateSyntax("foo")).toEqual("foo");
  });

  it("should return the error object for the invalid expression", () => {
    expect(validateSyntax("foo)(")).toMatchSnapshot();
  });
});
