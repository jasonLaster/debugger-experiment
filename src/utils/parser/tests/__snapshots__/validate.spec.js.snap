// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate syntax should return the error object for the invalid expression 1`] = `"SyntaxError : foo)( is an invalid."`;
